content/PreparedStatement/PreparedStatement.q.md

```diff
@@ -0,0 +1,1677 @@
+    );
+     
+    try {
+        $pstmt = $conn->prepare("INSERT INTO students (first_name, last_name, birth_date, gender) VALUES (?,?,?,?)");
+        foreach ($data as $row) {
+            if (!$pstmt->execute($row)) {
+                throw new Exception("Unable to insert values into the table.");
+            }
+        }
+        echo "Successfully inserted values into the table.";
+    } catch (Exception $e) {
+        echo $e->getMessage();
+    }
+?>
+```
+/// type=SS, answer=[5]
+
+Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Unable to establish a connection.`.
+
+- It prints `Successfully connected to the database.` and `Unable to insert values into the table.`.
+
+- It prints `Successfully connected to the database.` and `Successfully inserted values into the table.`.
+
+
+/// type=MS, answer=[4,5]
+
+Which statements correctly describe the code on line 9 of `connection.php`?
+
+- It evaluates the `$conn` object.
+
+- It sets the `PDO` class to the `PDO_PGQSL DSN` statement
+
+- It assigns the `PDO_PGQSL DSN` statement to the `PDO` class.
+
+- It creates the `$conn` object as an instance of the `PDO` class.
+
+- It passes the `PDO_PGQSL DSN` statement as an argument of the `PDO` class.
+
+
+/// type=SS, answer=[3]
+
+In `student.php`, what does the statement `require_once("connection.php");` do on line 2?
+
+- It updates the file `connection.php`.
+
+- It establishes a connection to the PostgreSQL database.
+
+- It includes the file `connection.php` in the file `student.php`.
+
+- It removes the file `connection.php` in the file `student.php`.
+
+- It excludes the file `connection.php` in the file `student.php`.
+
+
+/// type=MS, answer=[1,2,3,4,5]
+
+Which of the following are array values in the multidimensional array `$data`?
+
+- `Male`
+
+- `Berry`
+
+- `Vikki`
+
+- `Meisner`
+
+- `2002-09-09`
+
+
+/// type=MS, answer=[2,4]
+
+Which statements correctly describe the code on lines 4, 5, 6, and 7 of `student.php`?
+
+- It displays the elements of the multidimensional array `$data`.
+
+- It holds two arrays with four values in each key of `student1` and `student2`.
+
+- It removes the elements of the multidimensional array `$data` using the `array()` construct.
+
+- It creates the multidimensional array named `$data` with two keys `student1` and `student2`.
+
+- It accesses the elements of `$data` using the `array()` construct and the `[]` short array syntax.
+
+
+/// type=SS, answer=[3]
+
+In `student.php` what does `prepare()` method do on line 10?
+
+- It returns an array.
+
+- It executes the SQL statement.
+
+- It prepares the SQL statement for execution.
+
+- It contains the result set of the remaining rows.
+
+- It substitutes the input values of the SQL statement.
+
+
+/// type=SS, answer=[3]
+
+In `student.php`, what is the return value of `$conn->prepare("INSERT INTO students (first_name, last_name, birth_date, gender) VALUES (?,?,?,?)")` on line 10?
```

Comment:
For basic grammar error, nothing was detected in grammarly or any language tool I used for any problem in using `return value ` in the question.  But if you have run it in any language tool let me know.
`Return value - is the value you are expecting to be returned. `
But let me go ahead and make changes for your suggestion. Thanks.

Reply:
@redgeneLauren We have to always remember that we can not always rely on grammarly because we are dealing with a program and we all know that we asks the question after the program execution so it is common sense to identify that we have to use past tense for the verbs if we ask for the returned value. Hope this help. Thanks.